webapp/app/user/userManagement/edit/countrySelectionModal.js

```diff
@@ -2,68 +2,90 @@ import React from 'react'
 import * as R from 'ramda'
 
 import { Modal, ModalBody, ModalClose, ModalFooter, ModalHeader } from '@webapp/components/modal'
+import { useI18n } from '@webapp/components/hooks'
 
-class CountrySelectionModal extends React.Component {
-
-  render () {
-    const {
-      i18n,
-      countries,
-      userInfo, //session user
-      getCountryName,
-      toggleCountry,
-      onClose,
-      selection,
-      unselectableCountries,
-      headerLabel,
-    } = this.props
-
-    const countryRegions = R.groupBy(c => c.region, countries)
-    const isSelected = countryIso => R.contains(countryIso, selection)
-    const isUnselactable = countryIso => R.contains(countryIso, unselectableCountries)
-
-    return <div className="country-selection">
-      <Modal isOpen={true}>
+const CountrySelectionModalBody = (props) => {
+  const { countries, toggleCountry, selection, unselectableCountries } = props
+
+  const i18n = useI18n()
+
+  const isSelected = (countryIso) => R.contains(countryIso, selection)
+  const isUnselactable = (countryIso) => R.contains(countryIso, unselectableCountries)
+
+  // Sort given countries (from props) to hashmap: {regionCode}: [{countryIso},..]
+  const regionCountries = {}
+
+  countries.forEach((country) => {
+    const { countryIso, regionCodes } = country
+    regionCodes.forEach((regionCode) => {
+      if (!Array.isArray(regionCountries[regionCode])) {
+        regionCountries[regionCode] = []
+      }
+      regionCountries[regionCode].push(countryIso)
+    })
+  })
```

Comment:
wouldn't it be better to use useRegions here?

Reply:
This will be later fixed when the upper component gets some care